libbeat/processors/add_kubernetes_metadata/config.go

```diff
@@ -43,8 +46,52 @@ type Enabled struct {
 
 type PluginConfig []map[string]common.Config
 
+func trimAfter(value string, a string) string {
+	// Get substring after a string.
+	pos := strings.LastIndex(value, a)
+	if pos == -1 {
+		return ""
+	}
+	adjustedPos := pos + len(a)
+	if adjustedPos >= len(value) {
+		return ""
+	}
+	return value[adjustedPos:len(value)]
+}
+
+func findDaemonsConfigFlag() string {
+	// Search in processes and search for config in flags
+	psRes, _ := exec.Command("bash", "-c", "ps aux | grep -- -kubeconfig=").Output()
+	psResStr := string(psRes)
+	trimmedPsResStr := trimAfter(psResStr, "--kubeconfig=")
+	arr := strings.Split(trimmedPsResStr, " ")
+	kubeConfigPath := arr[0]
+	return kubeConfigPath
+}
+
+func getSystemKubeConfig() string {
+	homeKubeConfig := os.Getenv("HOME") + "/.kube/config"
+	if _, err := os.Stat(homeKubeConfig); !os.IsNotExist(err) {
+		return homeKubeConfig
+	}
+	envKubeConfig := os.Getenv("KUBECONFIG")
+	if _, err := os.Stat(envKubeConfig); !os.IsNotExist(err) {
+		return envKubeConfig
+	}
+	kubeConfigPath := findDaemonsConfigFlag()
+	if _, err := os.Stat(kubeConfigPath); !os.IsNotExist(err) {
+		return kubeConfigPath
+	}
+	kubeletConfig := "/etc/kubernetes/kubelet.conf"
+	if _, err := os.Stat(kubeletConfig); !os.IsNotExist(err) {
+		return kubeletConfig
+	}
+	return homeKubeConfig
```

Comment:
May we be missing the `InCluster` scenario?
https://github.com/kubernetes/client-go/blob/d4d906eb580f0bae1101160f0263cf8d2e7c8392/rest/config.go#L406-L440

Is there even a way of creating a `rest.Config`, and pass it along with the structure?
I would even think of creating the `ClientSet` from here if that makes sense.

Reply:
Nice catch. 
I can go with just getting the string out of the `InCluster()` result and continue with the current flow:
```
	clusterConfig, err := rest.InClusterConfig()
	if err != nil {
		return clusterConfig.String()
	}
```
I think this will be ok when it will be passed to `ClientSet` creation, so we avoid adding the extra filed in the struct and we keep the `ClientSet` creation as is without adding any casing inside its implementation. WDYT?

Reply:
After re-thinking of it, I conclude to pass this check in Kubernetes' client initialization. So if it is detected that we are `inCluster` the the clientset will be produced from ClusterConfig ignoring `KubeConfig` file. I think this is the safer way...

Reply:
That should work.

I think there is room for improving, but that involves modifying a bit the flow, and involves  `github.com/elastic/beats/libbeat/common/kubernetes`. I would say to move ahead with what you propose and we'll followup to tidy all this later.